### PR TITLE
Added crates to the non-alphabetical treated imports

### LINKED ISSUE #3118: crate::, self::, and super:: shouldn't be alphabetically sorted
In imports, `crate::`, `self::`, and `super::` imports shouldn't be reshuffled to appear in alphabetical order, since they're logically distinct from other imports (of external crates). For example, I write

```rust
use {
    crate::foo::bar,
    bytes::{Buf, BufMut},
    std::io,
};

mod foo {
    pub mod bar {}
}

fn main() {}
```

which rustfmt reorganizes to

```rust
use {
    bytes::{Buf, BufMut},
    crate::foo::bar,
    std::io,
};

mod foo {
    pub mod bar {}
}

fn main() {}
```

### PR DESCRIPTION
**Pull request by JoshBrudnak**
Added crates to list of imports that should not be alphabetized.

resolves #3118

### CODE REVIEW

**Comment by otavio** [comment]
Hello @JoshBrudnak, thanks for the pull request.

Is it possible for you to add a test case for it?

**Comment by nrc** [comment]
It looks like you're build is failing. Are you using a very up to date nightly compiler?

**Comment by otavio** [comment]
> It looks like you're build is failing. Are you using a very up to date nightly compiler?

My understanding is that the CI script is failing because the new `rustfmt` release does include some big changes on formatting and `cargo`'s code, used for integration test has not been updated. Am I mistaken?

File: src/imports.rs [hunk 1]
```diff
@@ -44,8 +44,7 @@ impl<'a> FmtVisitor<'a> {
             Some(item.vis.clone()),
             Some(item.span.lo()),
             Some(item.attrs.clone()),
-        )
-        .rewrite_top_level(&self.get_context(), shape);
+        ).rewrite_top_level(&self.get_context(), shape);
```
Comment (topecongiro) [inline src/imports.rs]: This (and other similar changes) should not happen. You could be using older rustfmt.
```suggestion
        )
        .rewrite_top_level(&self.get_context(), shape);
```

**Comment by JoshBrudnak** [comment]
I was using rustfmt stable by mistake.

**Comment by nrc** [comment]
@JoshBrudnak this looks good, thank you! Could you add a test please?

**Comment by nrc** [comment]
Awesome thank you!

I removed the source test file because if the target and source match, then you only need the target file.

**Pull request merged**